From bfee94b6657208dcde7e1ca500f7c4bdf5502a5d Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Mon, 7 Jul 2025 12:13:30 +0200
Subject: [PATCH] actions: include string to fix build with gcc-15

* fixes:
FAILED: [code=1] CMakeFiles/luneos_recovery_ui.dir/ui/actions.c.o
/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/recipe-sysroot-native/usr/bin/aarch64-webos-linux/aarch64-webos-linux-gcc --sysroot=/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/recipe-sysroot  -I/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/sources/luneos-recovery-ui-1.0/recovery/. -mbranch-protection=standard  --sysroot=/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/recipe-sysroot  -O2 -g  -ffile-prefix-map=/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/sources/luneos-recovery-ui-1.0/recovery=/usr/src/debug/luneos-recovery-ui/1.0  -ffile-prefix-map=/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/build=/usr/src/debug/luneos-recovery-ui/1.0  -ffile-prefix-map=/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/recipe-sysroot=  -ffile-prefix-map=/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/recipe-sysroot-native=  -pipe -DLV_CONF_INCLUDE_SIMPLE=1 -I/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/recipe-sysroot//usr/include/lvgl -MD -MT CMakeFiles/luneos_recovery_ui.dir/ui/actions.c.o -MF CMakeFiles/luneos_recovery_ui.dir/ui/actions.c.o.d -o CMakeFiles/luneos_recovery_ui.dir/ui/actions.c.o -c /OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/sources/luneos-recovery-ui-1.0/recovery/ui/actions.c
/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/sources/luneos-recovery-ui-1.0/recovery/ui/actions.c: In function 'event_handler_ChangeMassStorage':
/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/sources/luneos-recovery-ui-1.0/recovery/ui/actions.c:29:17: error: implicit declaration of function 'strcmp' [-Wimplicit-function-declaration]
   29 |         if(0 == strcmp(txt, "SDCard")) {
      |                 ^~~~~~
/OE/build/luneos-whinlatter/tmp/work/pinephone-webos-linux/luneos-recovery-ui/1.0/sources/luneos-recovery-ui-1.0/recovery/ui/actions.c:9:1: note: include '<string.h>' or provide a declaration of 'strcmp'
    8 | #include "lvgl/lvgl.h"
  +++ |+#include <string.h>
    9 |

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Submitted [https://github.com/Tofee/luneos-recovery-ui/pull/3]

 recovery/ui/actions.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/recovery/ui/actions.c b/recovery/ui/actions.c
index 631f4e5..8651fad 100644
--- a/recovery/ui/actions.c
+++ b/recovery/ui/actions.c
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <unistd.h>
 
 #include <sys/reboot.h>
